db/models.js

```diff
@@ -20,18 +21,36 @@ const getArticleData = async (id) => {
   const article = await checkExists("articles", "article_id", id);
   return article.rows[0];
 };
-const getArticlesData = async () => {
+const getArticlesData = async (
+  sort_by = "created_at",
+  order = "desc",
+  topic
+) => {
+  const validOrder = ["asc", "desc"];
+  
+  if (!validOrder.includes(order)) {
+    return Promise.reject({ status: 400, msg: "400 Bad Request" });
+  }
   try {
-    const articlesComments = await db.query(`SELECT articles.*, 
-  COUNT(comment_id) 
-  AS comment_count FROM articles 
-  LEFT JOIN comments ON comments.article_id = articles.article_id 
-  GROUP BY articles.article_id
-  ORDER BY created_at DESC`);
+    let column = sort_by;
+    let queryStr = format(
+      `SELECT articles.*, 
+    COUNT(comment_id) 
+    AS comment_count FROM articles 
+    LEFT JOIN comments ON comments.article_id = articles.article_id 
+    GROUP BY articles.article_id
+    ORDER BY %I`,
+      column
+    );
+    order === "asc" ? (queryStr += ` ASC`) : (queryStr += ` DESC`);
 
-    const articlesArray = articlesComments.rows
+    const articlesComments = await db.query(queryStr);
+    const articlesArray = articlesComments.rows;
     articlesArray.forEach((article) => delete article.body);
-    return articlesArray;
+    if (topic) {
+      await checkExists("articles", "topic", topic);
+      return articlesArray.filter((article) => article.topic === topic);
+    } else return articlesArray;
```

Comment:
Similarly here I would advise refactoring this a bit... 
- The function call for `checkExists` maybe should be in the controller in order to maintain the MVC pattern a bit better. Ideally only controllers should call model functions
- The operation this `filter` is doing should be done using _SQL_ because it can do this more efficiently than javascript plus it shows off more _SQL_ skills to any employers looking at this repo